src/macaron/slsa_analyzer/provenance/loader.py

```diff
@@ -103,9 +130,90 @@ def _load_provenance_file_content(
     if not isinstance(json_payload, dict):
         raise LoadIntotoAttestationError("The provenance payload is not a JSON object.")
 
+    if json_payload["predicate"]:
+        return json_payload
+
+    # For provenance without a predicate (e.g. PyPI), try to use the provenance certificate instead.
+    raw_certificate = json_extract(provenance, ["verification_material", "certificate"], str)
+    if not raw_certificate:
+        raise LoadIntotoAttestationError("Failed to extract certificate data.")
+    try:
+        decoded_certificate = base64.b64decode(raw_certificate)
+        certificate_claims = get_x509_certificate_values(decoded_certificate)
+    except UnicodeDecodeError as error:
+        raise LoadIntotoAttestationError("Cannot decode the payload.") from error
+    except ValueError as error:
+        logger.debug(error)
+        raise LoadIntotoAttestationError("Error parsing certificate.") from error
+
+    pypi_predicate = PyPICertificatePredicate.build_predicate(
+        certificate_claims["source_repo"],
+        certificate_claims["source_digest"],
+        certificate_claims["workflow"],
+        certificate_claims["invocation"],
+    )
+    json_payload["predicate"] = pypi_predicate
     return json_payload
 
 
+def get_x509_certificate_values(decoded_certificate: bytes) -> dict:
+    """Retrieve the values of interest from an x509 certificate.
+
+    Parameters
+    ----------
+    decoded_certificate: bytes
+        The decoded certificate bytes.
+
+    Returns
+    -------
+    dict
```

Comment:
Even though the return type of this function is a dictionary, it only contains a fix number of keys:
- source_repo
- source_digest
- workflow
- invocation
Do you think it would be worth capturing this in a `TypedDict`. Or instead we could return a instance of `PyPICertificatePredicate` from this function if the dictionary is not used elsewhere.